### PR TITLE
Adding dependency badging

### PR DESCRIPTION
**Pull request by ylil93**
also refactored dependency highlighter for usage in badge server

### CODE REVIEW

File: compatibility_lib/compatibility_lib/dependency_highlighter.py [hunk 1]
```diff
@@ -34,17 +34,17 @@ class UnstableReleaseError(Exception):
 
 
 class PriorityLevel(enum.Enum):
-    NOT_SET = "UP_TO_DATE"
-    LOW = "LOW_PRIORITY"
-    HIGH = "HIGH_PRIORITY"
+    UP_TO_DATE = 0
+    LOW = 1
+    HIGH = 2
```
Comment (ylil93) [inline compatibility_lib/compatibility_lib/dependency_highlighter.py]: Should I preserve "LOW_PRIORITY" and "HIGH_PRIORITY" as the enum names?

File: badge_server/badge_server.py [hunk 2]
```diff
@@ -120,6 +123,20 @@ def fake_redis_set(*args, **kwargs):
...
+    priority_level.LOW.name:          'black',
+    priority_level.HIGH.name:         'red'
+}
+
+DEP_CONVERSION_ERROR_RES = {
+    'status': 'CONVERSION_ERROR',
+    'details': None,
+}
+
+
```
Comment (liyanhui1228) [inline badge_server/badge_server.py]: Only one blank line is needed here.

File: badge_server/badge_server.py [hunk 3]
```diff
@@ -170,6 +187,17 @@ def _get_badge_url(version_and_res, package_name):
     return url
 
 
+def _get_dep_badge_url(res, package_name):
```
Comment (liyanhui1228) [inline badge_server/badge_server.py]: This could be combined with `_get_badge_url` above.

File: badge_server/badge_server.py [hunk 4]
```diff
@@ -279,6 +307,75 @@ def self_compatibility_badge_target():
...
+
+    package_name = flask.request.args.get('package')
+
+    res = {
+        'status': 'CALCULATING',
+        'details': None,
+    }
+
+    def run_check():
+        # First see if this package is already stored in BigQuery.
```
Comment (liyanhui1228) [inline badge_server/badge_server.py]: Remove this comment? Looks like accidentally copied.

File: compatibility_lib/compatibility_lib/dependency_highlighter.py (hunk 1 above)
Comment (liyanhui1228) [reply compatibility_lib/compatibility_lib/dependency_highlighter.py]: Yes that's good.

File: badge_server/badge_server.py [hunk 5]
```diff
@@ -279,6 +307,75 @@ def self_compatibility_badge_target():
...
+        outdated = highlighter.check_package(package_name)
+
+        max_level = priority_level.UP_TO_DATE
+        details = []
+        for dep in outdated:
+            level = dep.priority.level
+            if level.value > max_level.value:
```
Comment (liyanhui1228) [inline badge_server/badge_server.py]: Why having this? The details could just be `None` if the version is up to date.

Comment (ylil93) [reply badge_server/badge_server.py]: this snippet does 2 things:
1. track the highest outdated dependency (to be displayed on the badge)
2. append detail string for every outdated dependency to the list

if the list is empty, that means no outdated dependencies and the package is up to date and details is `EMPTY_DETAILS`

if the list is not empty, the list of strings are joined together to produce one string and provides info on all dependencies that are outdated

Comment (liyanhui1228) [reply badge_server/badge_server.py]: Sounds good, thanks for the explanation! And please also reply `done` or any comment you prefer to the rest of the review comments in future PRs, which makes it easier to track which were resolved. Thanks!

Comment (ylil93) [reply badge_server/badge_server.py]: ok! thanks for the review!

**Pull request merged**